dags/populate_metabase_fluxiae.py

```diff
@@ -91,50 +85,20 @@ def populate_fluxiae_referentials():
         populate_fluxiae_view(vue_name="fluxIAE_Salarie")
         populate_fluxiae_view(vue_name="fluxIAE_Structure")
 
-    def cleanup_import_directory(params=None, **kwargs):
-        flux_iae_import_directory = kwargs["ti"].xcom_pull(
-            task_ids="create_flux_iae_import_directory", key="work_directory"
-        )
-        shutil.rmtree(flux_iae_import_directory)
-
-    start = empty.EmptyOperator(task_id="start")
-
-    create_import_directory = python.PythonOperator(
-        task_id="create_flux_iae_import_directory",
-        provide_context=True,
-        python_callable=create_import_directory,
-    )
-
-    import_and_decrypt_flux_iae = python.PythonOperator(
-        task_id="import_and_decrypt_flux_iae",
-        provide_context=True,
-        python_callable=import_and_decrypt_flux_iae,
-    )
-
-    process_flux_iae = python.PythonOperator(
-        task_id="process_flux_iae",
-        provide_context=True,
-        python_callable=process_flux_iae,
-    )
-
-    cleanup_import_directory = python.PythonOperator(
-        task_id="import_flux_iae_cleanup", provide_context=True, python_callable=cleanup_import_directory
-    )
-
-    dbt_weekly = TriggerDagRunOperator(
-        task_id="trigger_dbt_weekly",
-        trigger_dag_id="dbt_weekly",
-        wait_for_completion=True,
-    )
-
-    end = slack.success_notifying_task()
+    @task(task_id="cleanup_import_directory")
+    def clean_work_directory(import_directory, **kwargs):
+        shutil.rmtree(import_directory)
 
+    work_directory = create_work_directory().as_setup()
```

Comment:
[Setup and Teardown](https://airflow.apache.org/docs/apache-airflow/stable/howto/setup-and-teardown.html); utile !